README.md

```diff
@@ -1,4 +1,274 @@
-Direct Embedding for Scala [![Build Status](https://travis-ci.org/directembedding/directembedding.png?branch=master)](https://travis-ci.org/directembedding/directembedding)
+DirectEmbedding [![Build Status](https://travis-ci.org/directembedding/directembedding.png?branch=master)](https://travis-ci.org/directembedding/directembedding)
 ===============
 
-A library for macro based direct embedding of DSLs.
+An experimental macro-based library for Embedded DSLs.
+
+# DirectEmbedding for DSLs
+
+
+DirectEmbedding is an experimental project that attempts to provide direct embedding in Scala for Domain Specific Languages (DSLs). Although, this project is currently exploratory, this library could become a very useful tool. For instance, it could replace the complex shadow embedding of [Yin-Yang](https://github.com/scala-yinyang/scala-yinyang/tree/cbfaf02405c5e273498c15ce943965eeba8afa31) library in existing or future projects such as [Slick](http://slick.typesafe.com/).
+
+This library provides **one** effortless logic for the reification of embedded DSLs in Scala. This comes along many **advantages**:
```

Comment:
This paragraph needs to be more specific. It does not require the DSL author to:
 - have knowledge of the Reflection API
 - take care of overloading resolution himself
...